### PR TITLE
watchmate: 0.4.5 -> 0.5.2-unstable-2024-08-13

### PR DESCRIPTION
**Pull request by aktaboot**
## Description of changes

Update watchmate, includes fix for Rust 1.80.0 breakage

Related #332957 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/misc/watchmate/default.nix [hunk 1]
```diff
@@ -10,22 +10,24 @@
 , wrapGAppsHook4
 , glib
 }:
-
+let appid = "io.gitlab.azymohliad.WatchMate";
+in
 rustPlatform.buildRustPackage rec {
   pname = "watchmate";
-  version = "0.4.5";
+  _version = "0.5.2";
```
Comment (alyssais) [inline pkgs/applications/misc/watchmate/default.nix:17]: Maybe move this to the `let`?

File: pkgs/applications/misc/watchmate/default.nix [hunk 2]
```diff
@@ -10,22 +10,24 @@
 , wrapGAppsHook4
 , glib
 }:
-
+let appid = "io.gitlab.azymohliad.WatchMate";
```
Comment (alyssais) [inline pkgs/applications/misc/watchmate/default.nix:13]: This doesn't seem likely to change, so why not leave it inlined?

Comment (aktaboot) [reply pkgs/applications/misc/watchmate/default.nix:13]: no real reason other than aesthetics tbh

File: pkgs/applications/misc/watchmate/default.nix [hunk 3]
```diff
@@ -10,22 +10,24 @@
 , wrapGAppsHook4
 , glib
 }:
-
+let 
+    _version = "0.5.2";
```
Comment (alyssais) [inline pkgs/applications/misc/watchmate/default.nix:14]: Indentation is off here.  `version'` might be a more idiomatic name.

Comment (alyssais) [reply pkgs/applications/misc/watchmate/default.nix:14]: Or `release` or `releaseVersion` if we want to avoid meaningful symbols.

Comment (aktaboot) [reply pkgs/applications/misc/watchmate/default.nix]: sure thing! thanks for the feedback :)

**Comment by aktaboot** [comment]
hmm, I should've split the formatting in a different commit I guess

**Comment by alyssais** [comment]
Yeah, please do. In the same commit it's quite hard to review.

**Pull request merged**